builtin/providers/google/resource_bigquery_dataset.go

```diff
@@ -0,0 +1,235 @@
+package google
+
+import (
+	"fmt"
+	"google.golang.org/api/bigquery/v2"
+	"google.golang.org/api/googleapi"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceBigQueryDataset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigQueryDatasetCreate,
+		Read:   resourceBigQueryDatasetRead,
+		Update: resourceBigQueryDatasetUpdate,
+		Delete: resourceBigQueryDatasetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"datasetId": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"friendlyName": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"defaultTableExpirationMs": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"access": &schema.Schema{
```

Comment:
This gets a bit tricky, since we implemented ACLs for GCS a little differently, but that was only possible because the API is different. The idea there was, you just provide a list of role entity pairs like [this](https://www.terraform.io/docs/providers/google/r/storage_bucket_acl.html). The GCS API basically squashed `userByEmail`, `groupByEmail`, and `domain` all into one field, so all we had to do was have the user concatenate the contents of that field with the `role`, and that would be the access control object. Second, there was no `view` type object attached with each GCS ACL, since the ACL is an entirely separate resource for GCS objects and buckets. Basically, to match the GCS Terraform interface, we'd really have to contort the BigQuery API, which is a shame. 

I'd say what you have here is best, I'm just leaving this comment in case people have questions about this decision in the future. 

@sparkprime, what do you think?